integration-tests/src/test/java/io/opentelemetry/B3PropagationIntegrationTest.java

```diff
@@ -20,67 +31,139 @@
 import io.opentelemetry.sdk.trace.data.SpanData;
 import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
 import java.io.IOException;
+import java.net.URI;
 import java.util.List;
-import java.util.Random;
+import java.util.function.Supplier;
 import java.util.stream.Stream;
 import org.jetbrains.annotations.Nullable;
 import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.RegisterExtension;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.ArgumentsProvider;
 import org.junit.jupiter.params.provider.ArgumentsSource;
-import org.testcontainers.shaded.okhttp3.Call;
-import org.testcontainers.shaded.okhttp3.OkHttpClient;
-import org.testcontainers.shaded.okhttp3.Request;
-import org.testcontainers.shaded.okhttp3.Response;
-import org.testcontainers.shaded.okhttp3.ResponseBody;
-import spark.Service;
 
 /** Integration tests for the B3 propagators, in various configurations. */
-public class B3PropagationIntegrationTest {
+class B3PropagationIntegrationTest {
 
-  private static final int server1Port = new Random().nextInt(40000) + 1024;
```

Comment:
While the probability is of course low, I did see a CI failure caused by both ports resolving to the same number which prompted this PR.